test/API/titerate.c

```diff
@@ -139,249 +141,248 @@ test_iter_group(hid_t fapl, hbool_t new_format)
     iter_info  info;                  /* Custom iteration information */
     H5G_info_t ginfo;                 /* Buffer for querying object's info */
     herr_t     ret;                   /* Generic return value */
-#else
     (void)fapl;
     (void)new_format;
-#endif
 
     /* Output message about test being performed */
     MESSAGE(
         5, ("Testing Group Iteration Functionality - SKIPPED for now due to no iteration restart support\n"));
-#ifndef NO_ITERATION_RESTART
-    /* Create the test file with the datasets */
-    file = H5Fcreate(DATAFILE, H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
-    CHECK(file, FAIL, "H5Fcreate");
+    if ((vol_cap_flags_g & H5VL_CAP_FLAG_ITERATE) && (vol_cap_flags_g & H5VL_CAP_FLAG_FILE_BASIC) &&
+        (vol_cap_flags_g & H5VL_CAP_FLAG_GROUP_BASIC)) {
```

Comment:
This particular test looks like it also need at least H5VL_CAP_FLAG_DATASET_BASIC, H5VL_CAP_FLAG_STORED_DATATYPES, H5VL_CAP_FLAG_GROUP_MORE and H5VL_CAP_FLAG_LINK_MORE, though we could also probably break the test down by parts.